src/main/java/org/primefaces/component/datatable/export/DataTableExporter.java

```diff
@@ -379,4 +384,24 @@ public int invoke(FacesContext context) {
             return counter;
         }
     }
+
+    protected void setResponseHeader(ExternalContext externalContext , String contentDisposition) {
+        externalContext.setResponseHeader("Expires", "0");
+        externalContext.setResponseHeader("Cache-Control", "must-revalidate, post-check=0, pre-check=0");
+        externalContext.setResponseHeader("Pragma", "public");
+        externalContext.setResponseHeader("Content-disposition", contentDisposition);
+    }
+
+    protected void addResponseCookie(ExternalContext externalContext) {
+        final boolean secure = ((HttpServletRequest) externalContext.getRequest()).isSecure();
+        if (secure) {
+            Map<String, Object> map = new HashMap<String, Object>();
```

Comment:
Initialize the HashMap size to 1 since you are only adding one value `new HashMap<>(1);`  You can just use the diamond operator as we are on Java 8